include/unifex/bind_back.hpp

```diff
@@ -26,16 +26,60 @@ namespace unifex {
 
 namespace _bind_back {
 
+namespace detail {
+
+struct compose {
+  template <typename Target, typename Other, typename Self>
+  auto operator()(Target&& target, Other&& other, Self&& self) const 
+    noexcept(
+      is_nothrow_callable_v<Other, Target> &&
+      is_nothrow_callable_v<Self, callable_result_t<Other, Target>>
+    ) 
+    -> callable_result_t<Self, callable_result_t<Other, Target>> {
+    return ((Self&&)self)(((Other&&)other)((Target&&)target));
+  }
+};
+
+} // namespace detail
+
+template <typename Cpo, typename Target>
+struct _apply_fn_impl {
+  struct type;
+};
+
 template <typename Cpo, typename... ArgN>
-struct _apply_target {
+using _apply_fn = typename _apply_fn_impl<Cpo, ArgN...>::type;
+
+template <typename Cpo, typename Target>
+struct _apply_fn_impl<Cpo, Target>::type {
+  std::remove_reference_t<Target>* target_;
```

Comment:
This is shorter and means the same thing:

```suggestion
  std::add_pointer_t<Target> target_;
```